### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Mzuha**

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Autowired
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallMapper cinemaHallMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallMapper = cinemaHallMapper;
+    }
+
+    @GetMapping
+    @ResponseBody
+    public List<CinemaHallResponseDto> get() {
```
Comment (EugeneSinica) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:33]:
```suggestion
    public List<CinemaHallResponseDto> getAll() {
```

**Review by EugeneSinica — APPROVED** [review]

**Review by vlasov-yurii — APPROVED** [review]

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    @ResponseBody
+    public MovieSessionResponseDto add(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = movieSessionMapper.toModel(movieSessionRequestDto);
+        MovieSession movieSessionFromDB = movieSessionService.add(movieSession);
+        return movieSessionMapper.toResponseDto(movieSessionFromDB);
+    }
+
+    @GetMapping("/available")
+    @ResponseBody
+    public List<MovieSessionResponseDto> getAvailableMovieSessions(@RequestParam Long movieId,
```
[the remark this answers is not in the document]
Comment (Mzuha) [reply src/main/java/mate/academy/spring/controller/MovieSessionController.java:46]: We do not receive all sesssions, but only the sessions we need.

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        return cinemaHallService.getAll().stream()
+                .map(cinemaHallMapper::toResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping
+    @ResponseBody
+    public CinemaHallResponseDto add(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
+        CinemaHall cinemaHall = cinemaHallMapper.toModel(cinemaHallRequestDto);
+        CinemaHall cinemaHallFromDB = cinemaHallService.add(cinemaHall);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:43]:
```suggestion
        CinemaHall cinemaHallFromDb = cinemaHallService.add(cinemaHall);
```
same everywhere

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    @GetMapping
+    @ResponseBody
+    public List<CinemaHallResponseDto> getAll() {
+        return cinemaHallService.getAll().stream()
+                .map(cinemaHallMapper::toResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping
+    @ResponseBody
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:40]: let's maybe just use `@RestController` everywhere?

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                                                                   @RequestParam LocalDate date) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:47]: you should be able to pass date in the following format: 29.02.2020

File: src/main/java/mate/academy/spring/dao/MovieSessionDao.java [hunk 6]
```diff
@@ -9,4 +9,8 @@
     List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
 
     Optional<MovieSession> get(Long id);
+
+    Optional<MovieSession> update(MovieSession newMovieSession);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/dao/MovieSessionDao.java:13]: why Optional?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java (hunk 5 above)
Comment (Mzuha) [reply src/main/java/mate/academy/spring/controller/MovieSessionController.java:47]: image
it works so

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**